renderdoc/driver/vulkan/vk_resources.cpp

```diff
@@ -3070,6 +3070,41 @@ void VkResourceRecord::MarkImageFrameReferenced(VkResourceRecord *img, const Ima
       id, maxRef, [](FrameRefType x, FrameRefType y) -> FrameRefType { return std::max(x, y); });
 }
 
+void VkResourceRecord::MarkImageViewFrameReferenced(VkResourceRecord *view, const ImageRange &range,
+                                                    FrameRefType refType)
+{
+  ResourceId img = view->baseResource;
+  ResourceId mem = view->baseResourceMem;
+
+  // mark image view as read
+  MarkResourceFrameReferenced(view->GetResourceID(), eFrameRef_Read);
+
+  // mark memory backing image as read
+  MarkResourceFrameReferenced(mem, eFrameRef_Read);
+
+  if(refType != eFrameRef_Read && refType != eFrameRef_None)
+    cmdInfo->dirtied.insert(img);
+
+  ImageRange imgRange;
+  imgRange.aspectMask = view->viewRange.aspectMask;
+  imgRange.baseMipLevel += view->viewRange.baseMipLevel + range.baseMipLevel;
+  imgRange.levelCount = range.levelCount;
+  imgRange.baseArrayLayer = view->viewRange.baseArrayLayer + range.baseArrayLayer;
+  imgRange.layerCount = range.layerCount;
+  imgRange.offset = range.offset;
+  imgRange.extent = range.extent;
+  imgRange.viewType = (view->viewRange.viewType <= 6) ? (VkImageViewType)view->viewRange.viewType
```

Comment:
I see that the 6 comes from the bit-packing limitations but I think it's liable to rot if a new image view type is added and the bitfield is expanded. It would be better to add an accessor that casts the bit-packed field to `VkImageViewType` and also checks for the maximum value and returns `VK_IMAGE_VIEW_TYPE_MAX_ENUM`, similar to the existing code that looks at `MipMaxValue`.

Reply:
I've added accessors